lib/raxx/http1.ex

```diff
@@ -0,0 +1,393 @@
+defmodule Raxx.HTTP1 do
+  @moduledoc """
+
+  ## Property testing
+
+  Functionality in this module might be a good opportunity for property based testing.
+  Elixir Outlaws convinced me to give it a try.
+
+  - Property of serialize then decode the head should end up with the same struct
+  - Propery of any number of splits in the binary should not change the output
+  """
+
+  @crlf "\r\n"
+
+  @doc """
+  # NOTE set_body should add content-length otherwise we don't know if to delete it to match on other end, when serializing
+
+  ## Examples
+      iex> request = Raxx.request(:GET, "http://example.com/path?qs")
+      ...> |> Raxx.set_header("accept", "text/plain")
+      ...> {head, _body} =  Raxx.HTTP1.serialize_request(request)
+      ...> :erlang.iolist_to_binary(head)
+      "GET /path?qs HTTP/1.1\\r\\nhost: example.com\\r\\naccept: text/plain\\r\\n\\r\\n"
+
+      iex> request = Raxx.request(:POST, "https://example.com/path")
+      ...> |> Raxx.set_header("content-type", "text/plain")
+      ...> |> Raxx.set_body(true)
+      ...> {head, _body} =  Raxx.HTTP1.serialize_request(request)
+      ...> :erlang.iolist_to_binary(head)
+      "POST /path HTTP/1.1\\r\\nhost: example.com\\r\\ntransfer-encoding: chunked\\r\\ncontent-type: text/plain\\r\\n\\r\\n"
+
+      iex> request = Raxx.request(:POST, "https://example.com/path")
+      ...> |> Raxx.set_header("content-length", "13")
+      ...> |> Raxx.set_body(true)
+      ...> {head, _body} =  Raxx.HTTP1.serialize_request(request)
+      ...> :erlang.iolist_to_binary(head)
+      "POST /path HTTP/1.1\\r\\nhost: example.com\\r\\ncontent-length: 13\\r\\n\\r\\n"
+  """
+  def serialize_request(request = %Raxx.Request{}) do
```

Comment:
add connect option, add note that could be extra headers in general case but i don't think want to tackle general case

Reply:
same for parsing and responses